Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/empty_dir/empty_dir.go

```diff
@@ -207,6 +203,47 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 	return ed.mounter.Mount("tmpfs", dir, "tmpfs", opts)
 }
 
+// setupDir creates the directory with the default permissions specified
+// by the perm constant, chmoding the directory if necessary to work around
+// the effective umask for the kubelet.
+func (ed *emptyDir) setupDir(dir string) error {
+	var err error
+	if err = os.MkdirAll(dir, perm); err != nil {
+		return err
+	}
+
+	fileinfo, err := os.Lstat(dir)
+	if err != nil {
+		return err
+	}
+
+	glog.V(1).Infof("Mode of %v: %v", dir, fileinfo.Mode())
+
+	if fileinfo.Mode().Perm() != perm.Perm() {
+		// If the permissions on the created directory are wrong, the
+		// kubelet is probably running with a umask set.  In order to
+		// avoid clearing the umask for the entire process or locking
+		// the thread, clearing the umask, creating the dir, restoring
+		// the umask, and unlocking the thread, we do a chmod to set
+		// the specific bits we need.
+		err := os.Chmod(dir, perm)
+		if err != nil {
+			return err
+		}
+
+		fileinfo, err = os.Lstat(dir)
+		if err != nil {
+			return err
+		}
+
+		if fileinfo.Mode().Perm() != perm.Perm() {
+			glog.Errorf("Unexpected mode for %v: %v", dir, fileinfo.Mode())
```

Comment:
Let's knock out https://github.com/pweil-/origin/commit/e09662d7281b832624f25738d986003cc2269b25#commitcomment-11634656 while we're in here so I won't have to do it for a follow up.